server/game/player.js

```diff
@@ -684,7 +684,14 @@ class Player extends GameObject {
         var baseCost = card.getCost();
         var matchingReducers = _.filter(this.costReducers, reducer => reducer.canReduce(playingType, card, target, ignoreType));
         var reducedCost = _.reduce(matchingReducers, (cost, reducer) => cost - reducer.getAmount(card, this), baseCost);
-        return Math.max(reducedCost, 0);
+        var costFloor = 0;
```

Comment:
isn't this just `return Math.max(reducedCost, ...matchingReducers.map(reducer => reducer.costFloor))`

Reply:
I don't think so, that would increase costs for cards that cost less than the floor.

Reply:
Hmmm, if that's not correct behaviour, then I think your code will do it wrong also.  Reducers can increase cost also.